hoomd/hpmc/compute.py

```diff
@@ -295,28 +348,80 @@ def _attach_hook(self):
     @log(category='sequence', requires_run=True)
     def sdf(self):
         """(*N_bins*,) `numpy.ndarray` of `float`): :math:`s[k]` - The scale \
-        distribution function :math:`[\\mathrm{probability\\ density}]`.
+        distribution function for compression moves \
+        :math:`[\\mathrm{probability\\ density}]`.
+
+        .. deprecated:: v3.8.0
+            Use `sdf_compression`.
+        """
+        self._cpp_obj.compute(self._simulation.timestep)
+        return self._cpp_obj.sdf_compression
+
+    @log(category='sequence', requires_run=True)
+    def sdf_compression(self):
+        """(*N_bins*,) `numpy.ndarray` of `float`): :math:`s[k]` - The scale \
+        distribution function for compression moves \
+        :math:`[\\mathrm{probability\\ density}]`.
 
         The :math:`x` at the center of bin :math:`k` is:
         :math:`x = k \\cdot \\delta x + \\delta x/2`.
 
         Attention:
             In MPI parallel execution, the array is available on rank 0 only.
-            `sdf` is `None` on ranks >= 1.
+            `sdf_compression` is `None` on ranks >= 1.
+        """
+        self._cpp_obj.compute(self._simulation.timestep)
+        return self._cpp_obj.sdf_compression
+
+    @log(category='sequence', requires_run=True)
+    def sdf_expansion(self):
+        """(*N_bins*,) `numpy.ndarray` of `float`): :math:`s[k]` - The scale \
+        distribution function for the expansion moves \
+        :math:`[\\mathrm{probability\\ density}]`.
+
+        The :math:`x` at the center of bin :math:`k` is:
+        :math:`x = -k \\cdot \\delta x - \\delta x/2`.
+
+        Attention:
+            In MPI parallel execution, the array is available on rank 0 only.
+            `sdf_expansion` is `None` on ranks >= 1.
         """
         self._cpp_obj.compute(self._simulation.timestep)
-        return self._cpp_obj.sdf
+        return self._cpp_obj.sdf_expansion
+
+    @log(category='sequence', requires_run=False)
+    def x_compression(self):
+        """(*N_bins*,) `numpy.ndarray` of `float`): The x \
+        values at the center of each bin corresponding to the scale \
+        distribution function for the compressive perturbations \
+        :math:`[\\mathrm{length}]`."""
+        sdf_compression = numpy.array(self.sdf_compression)
+        x = numpy.arange(0, len(sdf_compression), 1) * self.dx
```

Comment:
If it doesn't already exist, add a new C++ method to get the number of bins. Call that instead of `len(self.sdf_compression)`. This avoids making numerous extra copies of the histogram (see below).